### PR TITLE
Scale testing config

### PR DESCRIPTION
**Pull request by chrisstafford**

### CODE REVIEW

File: scale-testing/cluster-density/templates/secret.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
...
...RFVmMmtneXhvZ2FpOEFRQUEKQU1CV3FNbW9ERUhRbnlOdnlBT3lVellHV3NjelpvNm4vVTQvTkNvNlAzTzZCYnRWS0FUV1ZYWEVtNjVnZTBZTVRQYlJqWgpkWnE3d2hPV0h6cjVKRTNGanl4RllJWWEvdThPcVdjOEtzeUlIakZzM1dBT0t0RWI0L0x2WjFzWk5KT1I3OVVBYStud3ZqCm1rQXp6d3FxUTBzSFV0UWVoVHV0OTFNaURFNVB6OW8wdFZnQ25FbmJ1eGNuOEdRSmZONW1JZ0E2YnNzRDdmRWo3Sk03dnIKSE5ETjYzUjd2UTUyNTNNQ3lpUG93NW9iV01PcWZMU3VJVGxKNjg3NWJHQWNLQ2dNSUFBQURCQVA5M0l5MVFXcUptWnhsSQpueUZlU3g4Nk5meEJKV3hHc2l1MndqOE12ZldyWXQ4V2pQNHZjSEtSNjcydlc5UEs0Qk1GcDVNSVl2YkFnbGxhYnp2VVVyClF1VGhITWI3czlUL0JHM09yQUpyemVNZURrK3lybzlJaC9jRmxGS01xUGRoZzNLVEYxU1Jzb2F5L2pRVkh5QVIvY0ZnUmUKNzVaM3NQczNGOFJiSlpzL3VNUUxFbUJHVXAvNG9YbG1kKyt1c1FET2ptUWxsYm44bHh1N3gxbU42MDNzKytTVWVxWVVKZQo4cng5Y2hEU2tFTTJQWEhpdm9zbVUzdUZJaTRteUJBUUFBQU1FQTNGcURSQWliRVB3TGRqOWQ1UjJoTjI0eklmSGtadkEvClJxaDhzOCt5ZFUzakJQRGdGRUQwSGl4YXl0MVhnTzEwZDBGZEEvSFI2cE8xdUdybDdzdm5nYjdnSGJpTGo5SnBvRGd0SWkKalZnSEZvcmFZMWpTeGEyemNNbldFWVFaVFRlYUF6TU5qdWlJb2MwemN3ZHBRTWpkWWdwOUJBK1pVVEgvZTJNVmpmdUtmQQo3SlRDSW1uQUJlUWV1SWRtZ1VTZkltUTdRNGJsL29uUUZUZkVLc0tpZVFCU3hjS1VLa0orZ3R2RTJ5ZWJRRVFMV09kaE5GCjJ0S05TQlk5TUFsRTFsQUFBQUlISnpaWFpwYkd4aFFIZHZibVJsY214aGJtUXVjbk5sZG1sc2JHRXViM0puQVFJPQotLS0tLUVORCBPUEVOU1NIIFBSSVZBVEUgS0VZLS0tLS0K
```
Comment (naved001) [inline scale-testing/cluster-density/templates/secret.yml]: did you mean to commit this?

**Comment by chrisstafford** [comment]
yamlint failures are occurring due to kube-burners addition of golang template mechanics: https://pkg.go.dev/text/template

File: scale-testing/cluster-density/templates/secret.yml (hunk 1 above)
Comment (chrisstafford) [reply scale-testing/cluster-density/templates/secret.yml]: I did, this comes from kube-burner examples, I believe it's just a generic secret to add to the projects kube-burner creates. We should probably replace it with a generic secret though as I can generate a valid public key from the decoded private key.

Comment (chrisstafford) [reply scale-testing/cluster-density/templates/secret.yml]: I've edited the key to make it invalid and re-encoded it.

**Comment by joachimweyl** [comment]
@chrisstafford is the docs repo where you want this?

**Comment by chrisstafford** [comment]
> @chrisstafford is the docs repo where you want this?

@joachimweyl, I like it here. I also don't know where else it would go. In the future the kube-burner could be added as an application and the configs could be moved to nerc-ocp-app, but for now people can find the docs and required config files in one place.

**Review by naved001 — APPROVED** [review]
besides the CI failure, this looks fine to me.

**Comment by larsks** [comment]
> I also don't know where else it would go.

Would it make sense to have a separate repository for testing tools and configuration?

File: scale-testing/cluster-density/templates/secret.yml (hunk 1 above)
Comment (joachimweyl) [reply scale-testing/cluster-density/templates/secret.yml]: @naved001 can this issue be resolved?

**Comment by joachimweyl** [comment]
@larsks would you or @jtriley make the final decision on if we should have a separate repository for testing tools and configuration?

**Comment by joachimweyl** [comment]
@chrisstafford are you clear on what needs to change in order to pass failed checks?

**Comment by chrisstafford** [comment]
> @chrisstafford are you clear on what needs to change in order to pass failed checks?

> yamlint failures are occurring due to kube-burners addition of golang template mechanics: https://pkg.go.dev/text/template

I don't believe it will pass the yamlint checks due to the golang templating as mentioned above

**Comment by joachimweyl** [comment]
@chrisstafford is the plan to force it through or change the yamlint restrictions to work with golang?
@larsks do you have a preference about how we get golang to "pass" checks?

**Comment by naved001** [comment]
@joachimweyl @chrisstafford we can disable yamllint checks for that file by sticking a `yamllint disable-file` on top.

See: 
https://yamllint.readthedocs.io/en/stable/disable_with_comments.html#disabling-all-checks-for-a-file

```
naved@Naveds-MacBook-Pro docs % yamllint --strict scale-testing/cluster-density/templates/imagestream.yml
scale-testing/cluster-density/templates/imagestream.yml
  8:3       error    syntax error: could not find expected ':' (syntax)
```

after you make that change, the file won't be checked.

```diff
diff --git a/scale-testing/cluster-density/templates/imagestream.yml b/scale-testing/cluster-density/templates/imagestream.yml
index 43b1c4b..f045157 100644
--- a/scale-testing/cluster-density/templates/imagestream.yml
+++ b/scale-testing/cluster-density/templates/imagestream.yml
@@ -1,4 +1,4 @@
----
+# yamllint disable-file
 kind: ImageStream
 apiVersion: image.openshift.io/v1
 metadata:
@@ -7,3 +7,4 @@ spec:
 {{ if index . "image" }}
   dockerImageRepository: {{.image}}
 {{ end }}
+# yamllint enable
```

With that change in place:

```
naved@Naveds-MacBook-Pro docs % yamllint --strict scale-testing/cluster-density/templates/imagestream.yml
naved@Naveds-MacBook-Pro docs %
```

**Comment by larsks** [comment]
> @larsks would you or @jtriley make the final decision on if we should have a separate repository for testing tools and configuration?

@joachimweyl I would probably vote for putting this elsewhere (it's not really documentation), but as I'm going to be out for several days I'm not going to press for that.

W/r/t the `yamllint` checks, @naved001's suggestion is fine, or we could exclude the entire `scale-testing/cluster-density/templates` directory by updating the `.pre-commit-config.yaml`:

```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index d8b6455..d25bbba 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -22,5 +22,6 @@ repos:
     hooks:
       - id: yamllint
         files: \.(yaml|yml)$
+        exclude: ^scale-testing/cluster-density/templates/
         types: [file, yaml]
         entry: yamllint --strict
```

**Comment by joachimweyl** [comment]
@chrisstafford can you try one of the resolutions listed above to get this to pass the check?

**Comment by chrisstafford** [comment]
I've excluded the directory from yamllint, but I now agree with @larsks. Moving configs to their own repo would allow y'all to more simply employ pre-commit checks.

**Pull request merged**